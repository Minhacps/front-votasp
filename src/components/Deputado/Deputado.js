import React from 'react';

import './Deputado.scss';

const Deputado = ({
  name,
  number,
  politicalParty,
  matchScore,
  picture = 'http://via.placeholder.com/150x150?text=Foto',
}) => (
    <div className='deputado-list-item'>
      <div className='photo'>
        <img src={picture} alt={`Foto do candidato ${name}`} />
      </div>
      <div className='info'>
        <div>
          <span className='candidate-name'>{name}</span>
          <span className='candidate-number'>{number}</span>
        </div>

        <div>
          <span className='partido-name'>{politicalParty}</span>
        </div>

        <div>
          Afinidade: <span className='match-percent'>{matchScore / 100}%</span>
        </div>
      </div>
    </div>
  )

export default Deputado;
