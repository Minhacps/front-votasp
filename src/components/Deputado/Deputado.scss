.deputado-list-item {
  display: flex;
  border-bottom: 1px solid black;
  padding: 0px 0px;

  & > div {
    padding: 15px;
  }

  .photo img {
    width: 60px;
  }

  .photo > .photo-container {
    border-radius: 50%;
    width: 60px;
    height: 60px;
    overflow: hidden;
  }

  .info {
    flex-grow: 1;

    .candidate-name {
      font-weight: bold;
      text-transform: capitalize;
    }

    .candidate-number {
      margin-left: 10px;
    }

    .match-percent {
      border-radius: 40px;
      background-color: var(--roxoClaro);
      color: #fff;
      padding: 1px 20px;
      display: inline-block;
      font-size: var(--fontSize3);
    }
  }

  .buttons {
    display: flex;
    flex-direction: column;
    justify-content: space-around;

    button {
      background-color: var(--cinzaClaro);
      color: var(--cinza);
    }
  }
}
