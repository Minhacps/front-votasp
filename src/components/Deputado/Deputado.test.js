import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Deputado from './Deputado';
import CircleButton from '../CircleButton/CircleButton';

const deputado = {
  name: 'Nome do Candidato 1',
  number: 999,
  politicalParty: 'Nome Partido 1',
  matchScore: 8000
};

const props = {
  onClickPlus: jest.fn(),
  onClickShare: jest.fn(),
  ...deputado
};

describe('<Deputado />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Deputado {...props} />);
  });

  it('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
