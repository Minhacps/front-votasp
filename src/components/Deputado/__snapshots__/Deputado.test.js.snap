// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Deputado /> should match snapshot 1`] = `
<div
  className="deputado-list-item"
>
  <div
    className="photo"
  >
    <div
      className="photo-container"
    >
      <img
        alt="Foto do candidato Nome do Candidato 1"
        src="http://via.placeholder.com/150x150?text=Foto"
      />
    </div>
  </div>
  <div
    className="info"
  >
    <div>
      <a
        href="/perfil/undefined"
      >
        <span
          className="candidate-name"
        >
          Nome do Candidato 1
        </span>
      </a>
      <span
        className="candidate-number"
      >
        999
      </span>
    </div>
    <div>
      <span
        className="partido-name"
      >
        Nome Partido 1
      </span>
    </div>
    <div>
      Afinidade: 
      <span
        className="match-percent"
      >
        80
        %
      </span>
    </div>
  </div>
</div>
`;
